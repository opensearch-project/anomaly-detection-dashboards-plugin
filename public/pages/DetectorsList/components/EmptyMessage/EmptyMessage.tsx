/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { EuiButton, EuiEmptyPrompt, EuiText } from '@elastic/eui';
import React from 'react';
import { CreateDetectorButtons } from '../../../../components/CreateDetectorButtons/CreateDetectorButtons';

const FILTER_TEXT =
  'There are no detectors matching your applied filters. Reset your filters to view all detectors.';
const EMPTY_TEXT =
  'A detector is an individual anomaly detection task. You can create multiple detectors, ' +
  'and all the detectors can run simultaneously, with each analyzing data from different sources. ' +
  'Create an anomaly detector to get started.';

interface EmptyDetectorProps {
  isFilterApplied: boolean;
  onResetFilters: () => void;
}

export const EmptyDetectorMessage = (props: EmptyDetectorProps) => (
  <EuiEmptyPrompt
    style={{ maxWidth: '45em' }}
    body={
      <EuiText>
        <p>{props.isFilterApplied ? FILTER_TEXT : EMPTY_TEXT}</p>
      </EuiText>
    }
    actions={
      props.isFilterApplied ? (
        <EuiButton
          fill
          onClick={props.onResetFilters}
          data-test-subj="resetListFilters"
        >
          Reset filters
        </EuiButton>
      ) : (
        <CreateDetectorButtons />
      )
    }
  />
);
