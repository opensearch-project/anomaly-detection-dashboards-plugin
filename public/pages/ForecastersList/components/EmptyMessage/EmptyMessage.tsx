/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import { EuiSmallButton, EuiEmptyPrompt, EuiText, EuiLink } from '@elastic/eui';
import React from 'react';
import { CreateForecasterButtons } from '../../../../components/CreateForecasterButtons/CreateForecasterButtons';
import { FORECASTER_DOCS_LINK } from '../../../../utils/constants';

const FILTER_TEXT =
  'There are no results for your search. Reset the search criteria to view forecasts.';
const EMPTY_TEXT = 'Forecasts appear here.';

interface EmptyForecasterProps {
  isFilterApplied: boolean;
  onResetFilters: () => void;
}

export const EmptyForecasterMessage = (props: EmptyForecasterProps) => (
  <EuiEmptyPrompt
    data-test-subj="emptyForecasterListMessage"
    style={{ maxWidth: '45em' }}
    body={
      <EuiText size="s">
        <p>
          {props.isFilterApplied ? FILTER_TEXT : EMPTY_TEXT}
          {!props.isFilterApplied && (
            <>
              {' '}
              <EuiLink href={`${FORECASTER_DOCS_LINK}`} target="_blank">
                Learn more
              </EuiLink>
            </>
          )}
        </p>
      </EuiText>
    }
    actions={
      props.isFilterApplied ? (
        <EuiSmallButton
          fill
          onClick={props.onResetFilters}
          data-test-subj="resetListFilters"
        >
          Reset search criteria
        </EuiSmallButton>
      ) : (
        <CreateForecasterButtons />
      )
    }
  />
);
