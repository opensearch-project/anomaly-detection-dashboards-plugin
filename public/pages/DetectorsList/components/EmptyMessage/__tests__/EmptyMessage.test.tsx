/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { EmptyDetectorMessage } from '../EmptyMessage';

describe('<EmptyDetectorMessage /> spec', () => {
  describe('Empty results', () => {
    test('renders component with empty message', () => {
      const { container, getByText } = render(
        <EmptyDetectorMessage
          isFilterApplied={false}
          onResetFilters={jest.fn()}
        />
      );
      expect(container.firstChild).toMatchSnapshot();
      getByText('Create detector');
    });
  });
  describe('Filters results message', () => {
    test('renders component no result for filters message', () => {
      const { container } = render(
        <EmptyDetectorMessage
          isFilterApplied={true}
          onResetFilters={jest.fn()}
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test('resets filters when click on reset filters', () => {
      const handleResetFilters = jest.fn();
      const { getByTestId } = render(
        <EmptyDetectorMessage
          isFilterApplied={true}
          onResetFilters={handleResetFilters}
        />
      );
      fireEvent.click(getByTestId('resetListFilters'));
      expect(handleResetFilters).toHaveBeenCalled();
      expect(handleResetFilters).toHaveBeenCalledTimes(1);
    });
  });
});
