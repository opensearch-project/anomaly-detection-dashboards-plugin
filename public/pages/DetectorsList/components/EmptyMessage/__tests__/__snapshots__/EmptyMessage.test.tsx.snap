// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyDetectorMessage /> spec Empty results renders component with empty message 1`] = `
<div
  class="euiEmptyPrompt"
  style="max-width: 45em;"
>
  <span
    class="euiTextColor euiTextColor--subdued"
  >
    <div
      class="euiText euiText--medium"
    >
      <div
        class="euiText euiText--medium"
      >
        <p>
          A detector is an individual anomaly detection task. You can create multiple detectors, and all the detectors can run simultaneously, with each analyzing data from different sources. Create an anomaly detector to get started.
        </p>
      </div>
    </div>
  </span>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiSpacer euiSpacer--s"
  />
  <div
    class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--justifyContentCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <a
        class="euiButton euiButton--primary"
        data-test-subj="sampleDetectorButton"
        href="anomaly-detection-dashboards#/overview"
        rel="noreferrer"
        style="width: 200px;"
      >
        <span
          class="euiButtonContent euiButton__content"
        >
          <span
            class="euiButton__text"
          >
            Try a sample detector
          </span>
        </span>
      </a>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <a
        class="euiButton euiButton--primary euiButton--fill"
        data-test-subj="createDetectorButton"
        href="anomaly-detection-dashboards#/create-ad/"
        rel="noreferrer"
        style="width: 200px;"
      >
        <span
          class="euiButtonContent euiButton__content"
        >
          <span
            class="euiButton__text"
          >
            Create detector
          </span>
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`<EmptyDetectorMessage /> spec Filters results message renders component no result for filters message 1`] = `
<div
  class="euiEmptyPrompt"
  style="max-width: 45em;"
>
  <span
    class="euiTextColor euiTextColor--subdued"
  >
    <div
      class="euiText euiText--medium"
    >
      <div
        class="euiText euiText--medium"
      >
        <p>
          There are no detectors matching your applied filters. Reset your filters to view all detectors.
        </p>
      </div>
    </div>
  </span>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiSpacer euiSpacer--s"
  />
  <button
    class="euiButton euiButton--primary euiButton--fill"
    data-test-subj="resetListFilters"
    type="button"
  >
    <span
      class="euiButtonContent euiButton__content"
    >
      <span
        class="euiButton__text"
      >
        Reset filters
      </span>
    </span>
  </button>
</div>
`;
